thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -123,6 +118,20 @@ public DetectionTaskRunner() {
     return taskResult;
   }
 
+  protected void assertCompletenessCheck(String dataset) {
+    List<DataCompletenessConfigDTO> completed =
+        DAO_REGISTRY.getDataCompletenessConfigDAO().findAllByDatasetAndInTimeRangeAndStatus(
+            dataset, windowStart.getMillis(), windowEnd.getMillis(), true);
+
+    LOG.debug("Found {} dataCompleteness records for dataset {} from {} to {}",
+        completed.size(), dataset, windowStart.getMillis(), windowEnd.getMillis());
+
+    if (completed.size() <= 0) {
+      LOG.warn("Dataset {} is incomplete. Skipping anomaly detection.", dataset);
+      throw new IllegalStateException(String.format("Dataset %s incomplete", dataset));
```

Comment:
If this function does not interrupt the flow, then we can implement a feature like "Send an alert if the date quality of a certain dataset is bad for x times or for x hours."